### PR TITLE
Wrong behavior when executing `alter table t add index c (c(3));` (#2432)

### PR DESCRIPTION
**Pull request by idlesummerbreeze**
Hi @zimulala ,

PTAL

Regards

### CODE REVIEW

**Comment by shenli** [comment]
@idlesummerbreeze Thanks! Please solve the conflict.

**Comment by ngaut** [comment]
Good job, Please clean up commit messages.

File: util/types/etc.go [hunk 1]
```diff
@@ -49,10 +49,16 @@ func IsTypeChar(tp byte) bool {
 	}
 }
 
-//IsTypeSpecifiable returns a boolean indicating
-// whether the tp is a specifiable type.
-func IsTypeSpecifiable(tp byte) bool {
-	return IsTypeBlob(tp) || IsTypeChar(tp) || tp == mysql.TypeBit
+//IsTypePrefixable returns a boolean indicating
```
Comment (zimulala) [inline util/types/etc.go]: Please leave a blank after `//`. The same as line 58.
@idlesummerbreeze Thanks for your PR!

Comment (idlesummerbreeze) [inline util/types/etc.go]: Done

**Comment by zimulala** [comment]
LGTM

**Pull request merged**